### PR TITLE
texter assignment

### PR DESCRIPTION
**Pull request by sreynen**
For https://github.com/MoveOnOrg/Spoke/issues/304 (part of https://github.com/MoveOnOrg/Spoke/issues/277), adds texter assignment test. Also 
adds JOBS_SYNC flag, so tests can wait on related jobs (like texter assignment) to complete.

### CODE REVIEW

File: src/server/api/schema.js [hunk 1]
```diff
@@ -244,7 +245,7 @@ async function editCampaign(id, campaign, loaders, user) {
       payload: compressedString.toString('base64')
     })
     if (JOBS_SAME_PROCESS) {
-      uploadContacts(job).then()
+      uploadContacts(job)
```
Comment (schuyler1d) [inline src/server/api/schema.js:248]: we need to keep these `.then()`s same as below, at least for prod-ish environments.

Were you intending to complete this as:
```
if (JOBS_SAME_PROCESS) {
   (JOBS_SYNC ? await uploadContacts(job) : uploadContacts(job).then())
}
```

File: src/server/api/schema.js [hunk 2]
```diff
@@ -276,7 +277,12 @@ async function editCampaign(id, campaign, loaders, user) {
     })
 
     if (JOBS_SAME_PROCESS) {
-      assignTexters(job).then()
+      if (JOBS_SYNC) {
+        await assignTexters(job)
+      }
+      else {
+        assignTexters(job)
```
Comment (schuyler1d) [inline src/server/api/schema.js:284]: same for here, etc.

**Review by schuyler1d — APPROVED** [review]
need to bring back .then()s please update before merge.  if you restore them all then :+1:

**Review by schuyler1d — CHANGES_REQUESTED** [review]
changes needed

File: src/server/api/schema.js (hunk 1 above)
Comment (sreynen) [reply src/server/api/schema.js:248]: I believe these `.then()` calls aren't doing anything useful. I think they're only cloning the promise object returned by the async function, without changing it at all.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/then is the closest thing I could find to directly describing an empty `.then()` call:

> If one or both arguments are omitted, or are provided non-functions, then then will be missing the handler(s), but will not generate any errors.

**Review by schuyler1d — APPROVED** [review]
Ok, after testing this, @sreynen is right -- the `.then()` doesn't do anything.  sorry about the hangup
:+1:

**Pull request merged**